contracts/SportMarkets/Parlay/ParlayPolicy.sol

```diff
@@ -0,0 +1,145 @@
+import "../../interfaces/IParlayMarketsAMM.sol";
+
+import "../../interfaces/IParlayPolicy.sol";
+
+import "../../interfaces/ITherundownConsumer.sol";
+import "../../interfaces/IGamesPlayerProps.sol";
+
+contract ParlayPolicy is Initializable, ProxyOwned, ProxyPausable {
+    IParlayMarketsAMM public parlayMarketsAMM;
+    ISportsAMM public sportsAMM;
+    address public consumer;
+    mapping(uint => uint) public restrictedMarketsCount;
+    // toBeRemoved:
+    mapping(uint => bool) public isRestrictedToBeCombined;
+    mapping(uint => mapping(uint => bool)) public restrictedTagCombination;
+    mapping(bytes32 => mapping(uint => uint)) public restrictedTagComboCount;
+    mapping(uint => mapping(uint => bool)) public restrictedTag1Combo;
+    mapping(uint => uint) public maxPlayerPropsPerSport;
+
+    function initialize(address _owner, address _parlayMarketsAMM) external initializer {
+        setOwner(_owner);
+        parlayMarketsAMM = IParlayMarketsAMM(_parlayMarketsAMM);
+        sportsAMM = ISportsAMM(parlayMarketsAMM.sportsAmm());
+        consumer = sportsAMM.theRundownConsumer();
+    }
+
+    function areEligiblePropsMarkets(address _childMarket1, address _childMarket2)
+        external
+        view
+        returns (bool samePlayerDifferentProp)
+    {
+        samePlayerDifferentProp = IGamesPlayerProps(ITherundownConsumer(consumer).playerProps()).areEligiblePropsMarkets(
+            _childMarket1,
+            _childMarket2
+        );
+    }
+
+    function getSgpFeePerCombination(IParlayPolicy.SGPData memory params) external view returns (uint sgpFee) {
+        sgpFee = parlayMarketsAMM.getSgpFeePerCombination(
+            params.tag1,
+            params.tag2_1,
+            params.tag2_2,
+            params.position1,
+            params.position2
+        );
+    }
+
+    function getMarketDefaultOdds(address _sportMarket, uint _position) external view returns (uint odd) {
+        odd = sportsAMM.getMarketDefaultOdds(_sportMarket, false)[_position];
+    }
+
+    function isTags1ComboRestricted(uint tag1, uint tag2) external view returns (bool isRestricted) {
+        isRestricted = restrictedTag1Combo[tag1][tag2];
+    }
+
+    function isRestrictedComboEligible(
+        uint tag1,
+        uint tag2,
+        uint tag1Count,
+        uint tag2Count
+    ) external view returns (bool eligible) {
+        bytes32 tagHash = keccak256(abi.encode(tag1, tag2));
+        eligible = true;
+        uint restrictTag1 = restrictedTagComboCount[tagHash][tag1];
+        uint restrictTag2 = restrictedTagComboCount[tagHash][tag2];
+        if (restrictTag1 > 0 && restrictTag1 < tag1Count) {
+            eligible = false;
+        } else if (restrictTag2 > 0 && restrictTag2 < tag2Count) {
+            eligible = false;
+        }
+    }
+
+    function setMaxPlayerPropsPerSport(uint tag1, uint maxPlayerPropsGames) external onlyOwner {
+        maxPlayerPropsPerSport[tag1] = maxPlayerPropsGames;
```

Comment:
add checks for tags, also max player props, add it as variable `maxAllowedPlayerPropsGames > maxPlayerPropsGames`
event adding